src/messaging/messaging.ts

```diff
@@ -245,15 +301,26 @@ export class Messaging implements MessagingInterface {
       throw new FirebaseMessagingError(
         MessagingClientErrorCode.INVALID_ARGUMENT, 'dryRun must be a boolean');
     }
-    return this.getUrlPath()
-      .then((urlPath) => {
-        const request: { message: Message; validate_only?: boolean } = { message: copy };
-        if (dryRun) {
-          request.validate_only = true;
-        }
-        return this.messagingRequestHandler.invokeRequestHandler(FCM_SEND_HOST, urlPath, request);
+
+    const clientMessage = this.convertToClientMessage(copy);
+
+    const client = new FcmServiceClient({
```

Comment:
We should use the same `client` instance for multiple calls. You can do something like:

```ts
private getFcmServiceClient(): Promise<FcmServiceClient> {
  if (this.fcmServiceClient) {
    return Promise.resolve(this.fcmServiceClient);
  }

  return this.getClientOptions()
    .then((options) => {
      this.fcmServiceClient = new FcmServiceClient(options);
      return this.fcmServiceClient;
    });
}
```

Reply:
Agreed, added.